### PR TITLE
AAP-35593: add unit tests

### PR DESCRIPTION
**Pull request by jay-steurer**
Add unit tests to dvcs checks

### CODE REVIEW

File: test_check_dvcs.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        self.assertEqual(delete_previous_comment_if_needed("Deleting previous comment ... "))
+
+    def test_fail_delete_previous_comment(self):
+        self.assertEqual(delete_previous_comment_if_needed("Failed to delete previous comment"))
+
+class TestStringStartWithJira(unittest.TestCase):
+
+    def test_string_start_with_jira_ticket(self):
+        self.assertEqual(does_string_start_with_jira("JIRA-1234 This is a PR"), "JIRA-1234")
+
```
Comment (john-westcott-iv) [inline test_check_dvcs.py]: This would have to be `AAP-` right?

File: test_check_dvcs.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    def test_string_with_no_jira_ticket(self):
+        self.assertEqual(does_string_start_with_jira("NO-JIRA: Minor changes"), "NO-JIRA")
```
Comment (john-westcott-iv) [inline test_check_dvcs.py]: For this test I would see if we can do something like:
```python
from check_dvcs import _NO_JIRA_MARKER
self.assertEqual(does_string_start_with_jira(f"{_NO_JIRA_MARKER}: Minor changes"), _NO_JIRA_MARKER)

File: test_check_dvcs.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    def test_empty_string(self):
+        """Test with an empty string."""
+        self.assertIsNone(does_string_start_with_jira(""), "The PR string came back as None")
+
```
Comment (john-westcott-iv) [inline test_check_dvcs.py:31]: Lets add a test case for a random string as well. i.e. "This is some PR title that does not match".

**Review by john-westcott-iv — COMMENTED** [review]
We are likely going to need to add mocks for the requests calls to fake out the responses. We should also see how much code coverage we have. We might be able to implement SonarQube on this repo and generate a coverage report for this.

**Comment by jay-steurer** [comment]
Closing this PR in favor of  https://github.com/ansible/dvcs-action/pull/8

**Pull request closed without merging**